### DISCUSSION TITLE
SDR++ on a rasberry pi 4

### DISCUSSION THREAD

**Discussion by rsedaker** [AlexandreRouma/SDRPlusPlus#968] [Q&A]
At the onset, let me disclose that I am a noob to both linux and Rasberry Pi.

I want to run my SDRPlay RSPdx on my rasberry pi but I cannot get the APIs to install (or some are missing)
This is the error message where I get stuck:

"-- Checking for module 'libzstd'
--   No package 'libzstd' found
CMake Error at /usr/share/cmake-3.18/Modules/FindPkgConfig.cmake:545 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.18/Modules/FindPkgConfig.cmake:733 (_pkg_check_modules_internal)
  core/CMakeLists.txt:132 (pkg_check_modules)"
  
  Any and all help is appreciated.
  
  If you have SDR++ installed on your RasPi  could you please let me know how you did it?
  
  Thanks

**Comment by Terminator0007** [comment]
i can give you some advice to forget about RPi4B to use with SDR++ ... Installing of course should be ok without any problems if you do it all correctly but ... But I can recommend  in 100% to install on Odroid N2+ with Ubuntu. Even with 10MHz band with switched on all adds like a stereo, RDS, noise reduction CPU working max to 75% with 50C degree. Any problems with SDRPlay API or scanner or other functions. Raspberry Pi is an old useless device. I checked almost all SBC which are on the market now like Khadas VIM4, Radxa 5B, Orange Pi 5 and Odroid N2+ is for me number 1 now, very quiet and without fan . But if you want to still use RPi4 , clean the system, install 64bit Raspbian, install correctly SDR++ and any problems too but max to 6MHz wide band

**Comment by adevivo** [comment]
Instead of installing SDR++ on the Pi, you might want to install RTL-TCP (`sudo apt install rtl_tcp`). Then connect to that from android or some other system that you have SDR++ installed on.  From the client side (android or whatever) update the __Source__ in SDR++ to RTL-TCP and set your Pi's IP Address. This has worked well for me because I'm not tied to the antenna and can connect from anywhere.

**Reply by AlexandreRouma** [reply]
Why the hell would you use RTL-TCP when SDR++ has its own server mode which is better in every conceivable way...

**Comment by adevivo** [comment]
It's running on a Pi which IMO is underpowered for anything that really requires an interface. Also, as I mentioned, it remains available from anywhere not tied to the antenna or any other nonsense. I can connect using SDR++ from my workstation, android, or while I'm at the grocery store. The server in SDR++ might be great but I wouldn't know because I don't want to run my workstation 24 hours a day.

**Reply by AlexandreRouma** [reply]
SDR++ server doesn't require an interface... It's headless and doesn't use that much more CPU than rtl_tcp even though it supports compression, something you absolutely need over wifi or remote usage... And SDR++ for Android can connect to it just like a desktop, it doesn't matter. Please make sure you know what you're talking about before giving advice on here.

**Reply by bleez99** [reply]
Works fine on my Orange Pi 5 Plus running Ubuntu 22.04
Grabbed the code, chased down the build requirements, and I have my SDRPlay RSPdx serving up the entire house.
**_Seriously, why the hell would you ever turn the Pi off?_**
I have both the Pi and SDR++ running headless, Pi plugged into the router with ethernet for max bandwidth.

The Pi doesn't break a sweat, and the SDRPlay can be bandwidth hungry with samples as wide as 10Mhz 

<img width="939" alt="htop" src="https://github.com/AlexandreRouma/SDRPlusPlus/assets/3842064/e04d337a-7099-415d-94b3-4c129709a84b">

**Reply by bleez99** [reply]
Thank you @AlexandreRouma  for your fine work.
